ianmccurry11/TODO_List
# Monitor Task Completion and Record Task Completion

# Feature Overview:
## Implemented monitoring of task completion for the following time periods:
1. year
2. month
3. week
4. all-time
### User Metrics Monitoring Description
When a task is completed a new user metric entry if none has been previously added. Otherwise, the user metric is updated, and if the period of the current date no longer falls within the period of the user metrics the period is updated.
## Created user_metrics table
The user metrics table keeps track of the user owner and contains nested objects for user metrics for each time period.

### Testing (How to view feature)
0. Login/Register
1. Create a task
2. navigate to display tasks ```http://localhost:3000/display-tasks```
3. Set one of the tasks as completed
4. Navigate to the mongodb ```user_metrics``` and verify that the user_metrics was incremented/altered
